### PR TITLE
Feature/add checks and functionality for ajax submissions

### PR DESCRIPTION
**Pull request by codiebeulaine**

### CODE REVIEW

File: molo/surveys/models.py [hunk 1]
```diff
@@ -428,6 +432,17 @@ def serve(self, request, *args, **kwargs):
             form = self.get_form(request.POST, page=self, user=request.user)
```
Comment (moh-moola) [inline molo/surveys/models.py]: thinking all this `if request.method == 'POST':` logic block can be encapsulated within a "handle Post method" to keep the code legible

File: molo/surveys/models.py [hunk 2]
```diff
@@ -418,7 +418,11 @@ def has_page_breaks(self):
     def serve(self, request, *args, **kwargs):
         if (not self.allow_multiple_submissions_per_user and
                 self.has_user_submitted_survey(request, self.id)):
-            return render(request, self.template, self.get_context(request))
+            if request.method == 'POST' and 'ajax' in request.POST:
```
Comment (moh-moola) [inline molo/surveys/models.py]:
```
if request.method == 'POST' and 'ajax' in request.POST:
     pass
else:
     return render(
             request, self.template, self.get_context(request))
```
change to `if not (request.method == 'POST' and 'ajax' in request.POST)`

Comment (codiebeulaine) [reply molo/surveys/models.py]: this will fail if it is a get request though?

Comment (moh-moola) [reply molo/surveys/models.py]: Yes

**Pull request merged**